app/src/common/chromium/com/igalia/wolvic/browser/api/impl/YoutubeUrlHelper.java

```diff
@@ -0,0 +1,51 @@
+package com.igalia.wolvic.browser.api.impl;
+
+import org.chromium.url.GURL;
+
+import java.util.ArrayList;
+
+/**
+ * In addition to desktop UA YouTube requires the query parameter "app=desktop" to display 360
+ * videos correctly. This helper class implements the method to add this parameter to URL.
+ */
+public class YoutubeUrlHelper {
+    public static GURL maybeRewriteYoutubeURL(GURL url) {
+        if (!url.domainIs("youtube.com") && !url.domainIs("youtube-nocookie.com")) {
+            return url;
+        }
+
+        if (!url.getPath().equals("/watch")) {
+            return url;
+        }
+        
+        // Java version of GURL doesn't expose the same ReplaceComponents functionality as the
+        // native version, so we have to use string concatenation here.
+        return new GURL(url.getScheme() + "://" + url.getHost() + url.getPath() + "?" +
```

Comment:
Why not using Android's Uri class to properly construct it and then use the GURL(String) constructor just to return?